benchto-driver/src/main/java/io/trino/benchto/driver/execution/QueryExecutionDriver.java

```diff
@@ -61,12 +74,27 @@ private boolean isSelectQuery(String sql)
         return sql.startsWith("select") || sql.startsWith("show") || sql.startsWith("with");
     }
 
-    private QueryExecutionResult executeSelectQuery(Connection connection, QueryExecutionResultBuilder queryExecutionResultBuilder, String sqlStatement)
+    private QueryExecutionResult executeSelectQuery(
+            Connection connection,
+            QueryExecutionResultBuilder queryExecutionResultBuilder,
+            String sqlStatement,
+            Optional<Path> resultFile)
             throws SQLException
     {
         try (Statement statement = connection.createStatement();
                 ResultSet resultSet = statement.executeQuery(sqlStatement)) {
-            int rowsCount = fetchRows(sqlStatement, resultSet);
+            int rowsCount;
+            if (resultFile.isPresent()) {
+                if (Files.exists(resultFile.get())) {
```

Comment:
comparison is fine in benchto. what I meant was to have two jobs. one to set up the results - this could be run once (or once per new query set) and the other one that expects the results to be there (normal run).
it can be error-prone if we keep the `if (Files.exists(resultFile.get()))` because if the results are not there by mistake or the directory name is wrong we would not check the results (and what's worse we would upload the possible faulty results be accident)

Reply:
I'll add an option to decide if we require result file to exist.